AWS/aws-cloudtrail/ingest/parser.yml

```diff
@@ -0,0 +1,111 @@
+name: aws-cloudtrail
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+  - name: parsed_date
+    external:
+      name: date.parse
+      properties:
+        input_field: '{{json_event.message.eventTime}}'
+        output_field: datetime
+  - name: parsed_source
+    filter: '{{json_event.message.get("sourceIPAddress") != None}}'
+    external:
+      name: grok.match
+      properties:
+        input_field: json_event.message.sourceIPAddress
+        output_field: source
+        pattern: '(%{IP:ip}|%{HOSTNAME:domain})'
+  - name: set_ecs_fields
+  - name: set_action_fields
+  - name: set_aws_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          '@timestamp': '{{parsed_date.datetime}}'
+          event.kind: 'event'
+          event.category: ['network']
+          event.type: ['access']
+          event.dataset: 'cloudtrail'
+          event.action: '{{json_event.message.eventName or json_event.message.insightDetails.eventName}}'
+          event.code: '{{json_event.message.errorCode or json_event.message.insightDetails.errorCode}}'
+          event.reason: '{{json_event.message.errorMessage}}'
+          event.provider: '{{json_event.message.eventSource or json_event.message.insightDetails.eventSource}}'
+          cloud.provider: 'aws'
+          cloud.service.name: 'cloudtrail'
+          cloud.region: '{{json_event.message.awsRegion}}'
+          cloud.account.id: '{{json_event.message.userIdentity.accountId}}'
```

Comment:
Duplicated with user.id

Reply:
Applying ECS standard. The account id should landed in `cloud.account.id`